api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -425,85 +431,177 @@ public PatientIdentifierType getPatientIdentifierType(Integer patientIdentifierT
 	@SuppressWarnings("unchecked")
 	public List<Patient> getDuplicatePatientsByAttributes(List<String> attributes) {
 		List<Patient> patients = new Vector<Patient>();
-		
+		List<Integer> patientIds = new Vector<Integer>();
+
 		if (attributes.size() > 0) {
-			String select = "select distinct p1 from Patient p1, Patient p2";
-			String where = " where p1 <> p2 ";
-			String orderBy = " order by ";
-			
-			Class patient = Patient.class;
-			Set<String> patientFieldNames = new HashSet<String>(patient.getDeclaredFields().length);
```

Comment:
I actually don't seem to understand why we had this logic, seems like it can cause issues